src/coreclr/runtime-prereqs.proj

```diff
@@ -14,7 +14,7 @@
   <Import Project="$(RepositoryEngineeringDir)nativepgo.targets" />
 
   <ItemGroup>
-    <ProjectReference Include="tools\cdac-build-tool\cdac-build-tool.csproj" ReferenceOutputAssembly="false" />
+    <ProjectReference Condition="'$(ClrRuntimeSubset)' == 'true'" Include="tools\cdac-build-tool\cdac-build-tool.csproj" ReferenceOutputAssembly="false" />
```

Comment:
There are still some twists and turns in subset mappings to get these right but we can use this, otherwise unused, property:
```suggestion
    <ProjectReference Condition="'$(ClrDefaultRuntimeBuildSubsets)' != ''" Include="tools\cdac-build-tool\cdac-build-tool.csproj" ReferenceOutputAssembly="false" />
```

Reply:
What would this do? From my reading we set `ClrDefaultRuntimeBuildSubsets` unconditionally. I'm mostly trying to avoid building this for `build.sh clr.iltools` which is what ILLinker testing needs and doesn't need cdac-build-tool (but currently builds it).